backend/main.py

```
@@ -103,22 +108,53 @@ def read_root():
+    """
+    Trigger clustering for the given project, preferring asynchronous fire-and-forget execution.
+    
+    If an asyncio event loop is running, schedules a non-blocking clustering job for the project.
+    If no event loop is available (e.g., in test environments), runs clustering synchronously so callers can observe results. Any exceptions raised during inline execution are logged.
+    
+    Parameters:
+        project_id (str): Identifier of the project whose unclustered feedback should be clustered.
+    """
+    try:
+        loop = asyncio.get_running_loop()
+        loop.create_task(maybe_start_clustering(project_id))
```

Ruff (0.14.8)

Store a reference to the return value of `loop.create_task`

(RUF006)